src/wallet/wallet_rpc_server.cpp

```diff
@@ -2424,12 +2434,34 @@ namespace tools
       er.message = "Failed to open wallet";
       return false;
     }
-    if (m_wallet)
-      delete m_wallet;
     m_wallet = wal.release();
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_close_wallet(const wallet_rpc::COMMAND_RPC_CLOSE_WALLET::request& req, wallet_rpc::COMMAND_RPC_CLOSE_WALLET::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_wallet) return not_open(er);
+    if (m_wallet->restricted())
+    {
+      er.code = WALLET_RPC_ERROR_CODE_DENIED;
+      er.message = "Command unavailable in restricted mode.";
+      return false;
+    }
+
+    try
+    {
+      m_wallet->store();
```

Comment:
I don't think you need a store.

Reply:
```on_stop_wallet``` calls ```m_wallet->store()```, so I did the same thing here.

Are we sure that the wallet cache file would be up to date if we deleted ```m_wallet``` without doing a ```store()``` first?

Reply:
There are multiple places where it is not called though. The cache would be updated as long as the server refresh loop is running, which it would be.

Reply:
@jtgrassie @glv2 

> The cache would be updated as long as the server refresh loop is running, which it would be.

The thing I've faced with v.0.12.2.0 is that opening the other wallet or stopping the wallet with the `stop_wallet` method results to the `tx_key`s collected during transfer creations being lost. But if I do call the `store` method before stopping or switching the wallet the private keys of outgoing transfers persisted as expected.
IHMO, `store()` need to be done within `close_wallet` method.